packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -366,13 +366,21 @@ export class Summarizer extends EventEmitter implements ISummarizer {
             try {
                 const ack = await this.summaryCollection.waitSummaryAck(refSequenceNumber);
                 refSequenceNumber = ack.summaryOp.referenceSequenceNumber;
-
-                await this.internalsProvider.refreshLatestSummaryAck(
-                    ack.summaryOp.contents.handle,
-                    ack.summaryAck.contents.handle,
-                    refSequenceNumber,
-                    summaryLogger,
-                );
+                this.runningSummarizer.lockedRefreshSummaryAckAction(async () => {
+                    await this.internalsProvider.refreshLatestSummaryAck(
+                        ack.summaryOp.contents.handle,
+                        ack.summaryAck.contents.handle,
+                        refSequenceNumber,
+                        summaryLogger,
+                    );
+                }).catch((error) => {
+                    this.logger.sendErrorEvent({
```

Comment:
Not sure why we need one extra try/catch/log layers - HandleSummaryAckError even below is good enough for that, no? If you want to record handles, you can move definition of `ack` variable up outside of loop, and reset its state in the beginning of the loop, thus making it possible to report values (undefined if we exception happens from waitSummaryAck) as part of HandleSummaryAckError event